dynosaur_derive/src/expand.rs

```diff
@@ -148,6 +138,19 @@ pub(crate) fn expand_sig_ret_ty_to_pin_box(sig: &mut Signature) {
     sig.output = parse_quote! { #arrow ::core::pin::Pin<Box<dyn #ret + 'dynosaur>> };
 }
 
+pub(crate) fn expand_sig_ret_ty_to_box(sig: &mut Signature) {
+    let (arrow, ret) = expand_arrow_ret_ty(sig);
+    if let Some(asyncness) = sig.asyncness.take() {
+        sig.fn_token.span = asyncness.span;
+    }
+
+    if has_lifetime_bound(sig) {
```

Comment:
Seems like a mild sign of inconsistent layering that the caller also checks this condition before calling the function, but it's okay to leave it.

Reply:
The caller doesn't check to call this function, it checks before to call it, to see if we need to expand input args or not. This is about the return type.
Anyway it may need to be structured a little but better maybe 😁

Reply:
I realize that, I just think it's a smell that it gets checked at two different levels of abstraction. Not a big deal though, I know proc macros can get messy like this!

Reply:
This issue doesn't exist anymore and neither this function. The code is simpler now.